File: static/js/caesar.js
// caesar.js — İstemci tarafı Caesar Cipher (kaydırmalı şifre) mantığı.
//
// Kaydırma (shift) değeri backend tarafından oda kurulurken üretilir
// (bkz. ciphers/caesar.py -> generate_caesar_shift) ve odaya katılan
// istemciye join_confirmed olayıyla gönderilir (data.shift).

ALPHABET = (
   ' 0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZÇĞİÖŞÜabcdefghijklmnopqrstuvwxyzçğıöşü.,!?:;\'"-()\n'
)


function caesarEncrypt(text, shift) {
    let result = "";
    for (let char of text) {
        let index = ALPHABET.indexOf(char);
        result += index === -1 ? char : ALPHABET[(index + shift) % ALPHABET.length];
    }
    return result;
}

function caesarDecrypt(text, shift) {
    let result = "";
    for (let char of text) {
        let index = ALPHABET.indexOf(char);
        if (index === -1) {
            result += char;
        } else {
            let newIndex = index - shift;
            if (newIndex < 0) newIndex += ALPHABET.length;
            result += ALPHABET[newIndex];
        }
    }
    return result;
}

File: ciphers/caesar.py
"""
Caesar Cipher (kaydırmalı şifre) mantığı.
Her oda için rastgele bir 'shift' değeri üretilir; bu değer odanın
gizli anahtarı gibi davranır ve sadece odaya bağlanan istemcilere
(join_confirmed üzerinden) gönderilir.
"""

import os

ALPHABET = (
   ' 0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZÇĞİÖŞÜabcdefghijklmnopqrstuvwxyzçğıöşü.,!?:;\'"-()\n'
)


def generate_caesar_shift() -> int:
    """1 ile len(ALPHABET)-1 arasında rastgele bir kaydırma değeri üretir."""
    return (os.urandom(1)[0] % (len(ALPHABET) - 1)) + 1


def caesar_encrypt(text: str, shift: int) -> str:
    cipher = ""
    for c in text:
        index = ALPHABET.find(c)
        if index == -1:
            cipher += c
        else:
            cipher += ALPHABET[(index + shift) % len(ALPHABET)]
    return cipher


def caesar_decrypt(text: str, shift: int) -> str:
    return caesar_encrypt(text, -shift)
